src/libs/typesystem/specelektra/Elektra/Parse.hs

```diff
@@ -0,0 +1,35 @@
+--
+-- @file
+--
+-- @brief Parses GHC's internal type reprentation to an intermediate data structure and vice versa
+--
+-- @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+-- 
+module Elektra.Parse (fromAST, toAST) where
+
+import Data.Maybe    (fromMaybe)
+import Control.Monad (mapM, join)
+
+import Elektra.Types
+import qualified FiniteAutomata as FA
+
+import GhcPlugins
+import TyCoRep (Type(..), TyLit(..))
+
+-- This also ensures that every regex expression is a valid regex otherwise it returns EmptyRegex
```

Comment:
What does "This" refer to? Why "also"? What else is ensured?

Reply:
this is the function in general, but also is unnecessary it doesn't do more than that )
also i don't normalize here anymore only during the constraint resolution that is a relict from experimentation as well